src/python/WMCore/Storage/StageOutMgr.py

```diff
@@ -211,141 +235,137 @@ def __call__(self, fileToStage):
+                    logging.info("======>  %s\n", fileToStage['LFN'])
+                    logging.info("======>  %s\n using this stage out", stageOutStr(stageOut_rfc[0]))
+                    fileToStage = stageoutPolicyReport(fileToStage,stageOut_rfc[0]['phedex-node'],
+                                                      stageOut_rfc[0]['command'], 'LOCAL', 60311)
+                    continue
+                except Exception as ex:
+                    lastException = StageOutFailure("Error during local stage out",
+                                                    error=str(ex))
+                    logging.info("===> Stage Out Failure for file:\n")
+                    logging.info("======>  %s\n", fileToStage['LFN'])
+                    logging.info("======>  %s\n using this stage out", stageOutStr(stageOut_rfc[0]))
+                    fileToStage = stageoutPolicyReport(fileToStage, stageOut_rfc[0]['phedex-node'],
+                                                      stageOut_rfc[0]['command'], 'LOCAL', 60311)
+                    continue
+        
+        else:
+            logging.info("===> Attempting stage outs from override")
             try:
-                pfn = self.fallbackStageOut(lfn, fileToStage['PFN'],
-                                            fallback, fileToStage.get('Checksums'))
+                pfn = self.stageOut(lfn, fileToStage['PFN'], fileToStage.get('Checksums'))
                 fileToStage['PFN'] = pfn
-                fileToStage['PNN'] = fallback['phedex-node']
-                fileToStage['StageOutCommand'] = fallback['command']
-                logging.info("attempting fallback")
+                fileToStage['PNN'] = self.overrideConf['phedex-node']
+                fileToStage['StageOutCommand'] = self.overrideConf['command']
+                logging.info("attempting override stage out")
                 self.completedFiles[fileToStage['LFN']] = fileToStage
                 if lfn in self.failed:
                     del self.failed[lfn]
 
                 logging.info("===> Stage Out Successful: %s", fileToStage)
-                fileToStage = stageoutPolicyReport(fileToStage, None, None, 'FALLBACK', 0)
+                fileToStage = stageoutPolicyReport(fileToStage, None, None, 'OVERRIDE', 0)
                 return fileToStage
             except Exception as ex:
-                fileToStage = stageoutPolicyReport(fileToStage, fallback.get('phedex-node', None),
-                                                   fallback['command'], 'FALLBACK', 60310)
+                fileToStage = stageoutPolicyReport(fileToStage, self.overrideConf['phedex-node'],\
+                    self.overrideConf['command'], 'OVERRIDE', 60310)
                 lastException = ex
-                continue
 
         raise lastException
-
-    def fallbackStageOut(self, lfn, localPfn, fbParams, checksums):
+    
+    
+    def stageOut(self, lfn, localPfn, checksums, stageOut_rfc=None):
```

Comment:
Again for my education, my understanding is that there is no more definition of local and fallback stage out. Is that correct? I guess it's all part of this `stageOut` method now?

Reply:
Yes, exactly